pymc/gp/cov.py

```diff
@@ -529,23 +533,19 @@ def __init__(
                 ls = 1.0 / ls_inv
         self.ls = pt.as_tensor_variable(ls)
 
-    def square_dist(self, X, Xs):
-        X = pt.mul(X, 1.0 / self.ls)
+    @staticmethod
+    def square_dist(X, Xs, ls):
+        X = pt.mul(X, 1.0 / ls)
         X2 = pt.sum(pt.square(X), 1)
-        if Xs is None:
-            sqd = -2.0 * pt.dot(X, pt.transpose(X)) + (
-                pt.reshape(X2, (-1, 1)) + pt.reshape(X2, (1, -1))
-            )
-        else:
-            Xs = pt.mul(Xs, 1.0 / self.ls)
-            Xs2 = pt.sum(pt.square(Xs), 1)
-            sqd = -2.0 * pt.dot(X, pt.transpose(Xs)) + (
-                pt.reshape(X2, (-1, 1)) + pt.reshape(Xs2, (1, -1))
-            )
+        Xs = pt.mul(Xs, 1.0 / ls)
```

Comment:
Why did you take out the `is None` case? Can't you just do `Xs = X if Xs is None else Xs` and have a single branch?

Reply:
I moved it down to line 564 (and your one-liner is nicer).   I moved it because I thought it'd be easier for someone making a different distance function if the inputs didn't have defaults.